lcm-cmake/cpack.cmake

```diff
@@ -0,0 +1,68 @@
+# Package release version
+set(PACKAGE_RELEASE_VERSION 1)
+
+# Default CPack generators
+set(CPACK_GENERATOR TGZ STGZ)
+
+# Detect OS type, OS variant and target architecture
+if(UNIX)
+    set(OS_TYPE Linux)
+
+    # Determine distribution
+    execute_process(COMMAND lsb_release -si
+        OUTPUT_VARIABLE LINUX_DISTRO
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    # Determine architecture
+    execute_process(COMMAND uname -m
+        OUTPUT_VARIABLE MACHINE_ARCH
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    # Add CPack generator according to detected distribution
+    if(LINUX_DISTRO STREQUAL "Debian" OR LINUX_DISTRO STREQUAL "Ubuntu" OR LINUX_DISTRO STREQUAL "LinuxMint")
+        list(APPEND CPACK_GENERATOR DEB)
+    endif()
+
+    # Set OS type and ARCH suffix
+    set(OS_TYPE_ARCH_SUFFIX ${OS_TYPE}-${MACHINE_ARCH})
+else()
+    if(WIN32)
+        set(OS_TYPE Win)
+    elseif(APPLE)
```

Comment:
Er... no, as I was saying, MacOS is a `UNIX`, so you'll never get here. You want something more like:

```
if(UNIX)
  if(APPLE)
    ...MacOS stuff...
  else()
    ...lsb_release stuff...
    ...maybe add DEB...
  endif()
  ...uname stuff...
elseif(WIN32)
  ...Windows stuff...
else()
  ...error?...
endif()
```

I'm not sure the error is needed; I'm not sure CMake supports any platforms that are neither `UNIX` nor `WIN32`. Also, note that these days you probably want `MacOS`, not `OSX` or variants thereon.